src/group_theory/group_action.lean

```diff
@@ -21,16 +21,25 @@ infixr ` • `:73 := has_scalar.smul
 (one_smul : ∀ b : β, (1 : α) • b = b)
 (mul_smul : ∀ (x y : α) (b : β), (x * y) • b = x • y • b)
 
+/-- A typeclass mixin saying that two actions on the same space commute. -/
+class smul_comm_class (M N α : Type*) [has_scalar M α] [has_scalar N α] : Prop :=
+(smul_comm : ∀ (a : M) (a' : N) (b : α), a • a' • b = a' • a • b)
+
+export mul_action (mul_smul) smul_comm_class (smul_comm)
+
+lemma smul_comm_class.symm (M N α : Type*) [has_scalar M α] [has_scalar N α]
+  [smul_comm_class M N α] : smul_comm_class N M α :=
+⟨λ a' a b, (smul_comm a a' b).symm⟩
```

Comment:
I assume this can't be an instance? It might be nice to stick a comment on it to that effect.

Reply:
Added a docstring.